internal/connector/connector.go

```diff
@@ -270,76 +270,79 @@ func httpTransportFromOptions(opts ServerOptions) *http.Transport {
 	return transport
 }
 
-func syncWithServer(con connector) func(context.Context) {
-	return func(context.Context) {
-		host, port, err := con.k8s.Endpoint()
-		if err != nil {
-			logging.Errorf("failed to lookup endpoint: %v", err)
-			return
-		}
+func syncDestination(con connector) error {
+	host, port, err := con.k8s.Endpoint()
```

Comment:
The next step for allowing this to run locally will be to add a config setting that allows us to skip the endpoint lookup. So before making that change I'm extracting this logic from `sycnWithServer` to give us some space to add that new logic.